### DISCUSSION TITLE
Backup of keys used for backup encryption

### DISCUSSION THREAD

**Discussion by amgonzalezf** [cloudnative-pg/cloudnative-pg#2372] [General]
Hi,

I enabled encryption of the data and wal files of the backups:

spec:
  backup:
    barmanObjectStore:
      data:
        encryption: AES256
      wal:
        encryption: AES256

But I'm wondering what would happen if the kubernetes cluster where I'm running the database fails and we must create a new kubernetes cluster from scratch and restore those backups. I guess that in that scenario, I would lose the key used to encrypt and decrypt those backups, and therefore, they would become useless.

I'm thinking about making a copy of that key and storing it somewhere safely, so I can retrieve it and use it in this scenario. So here are my questions:

- How can I access this key to make a copy?
- Is the key automatically rotated?
- If I create a new kubernetes cluster from scratch, how could I restore that encryption key so the operator can use it to decrypt the backups?

**Comment by maxpain** [comment]
The encryption option will just inform the S3 bucket which method to use to store the data encrypted. Encrypted data cannot be read by any other AWS user but the owner of the bucket. Barman cloud does not encrypt any object before sending it to S3, it just asks the bucket to store them encrypted if S3 has been properly configured.
